File: blog-admin/blog_admin/blog_admin/models/page_translation.py
from django.db import models


class PageTranslation(models.Model):
    TRANSLATION_STATUS_CHOICES = [
        ('draft', 'Draft'),
        ('in_progress', 'In Progress'),
        ('ready', 'Ready'),
        ('published', 'Published'),
    ]

    page = models.ForeignKey(
        'Page', on_delete=models.CASCADE, related_name='translations'
    )
    language = models.ForeignKey(
        'Language',
        on_delete=models.CASCADE,
        db_column='language_code',
        to_field='code',
    )
    title = models.TextField()
    content = models.TextField(blank=True, null=True)
    seo_title = models.CharField(max_length=255, blank=True, null=True)
    seo_description = models.TextField(blank=True, null=True)
    translation_status = models.CharField(
        max_length=20, choices=TRANSLATION_STATUS_CHOICES, default='draft'
    )
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        managed = False
        db_table = 'page_translations'
        unique_together = [('page', 'language')]

    def __str__(self):
        return f"{self.title} [{self.language_id}]"

File: blog-api/src/main/scala/su/wps/blog/models/domain/PageTranslation.scala
package su.wps.blog.models.domain

import io.circe.Encoder

import java.time.ZonedDateTime

final case class PageTranslation(
  pageId: PageId,
  languageCode: String,
  title: String,
  content: Option[String],
  seoTitle: Option[String],
  seoDescription: Option[String],
  translationStatus: String,
  createdAt: ZonedDateTime,
  updatedAt: ZonedDateTime,
  id: Option[PageTranslationId] = None
)

final case class PageTranslationId(value: Int) extends AnyVal

object PageTranslationId {
  implicit val encoder: Encoder[PageTranslationId] = Encoder[Int].contramap(_.value)
}
